### PR TITLE
bpf: Use "fallthrough;", compile with -Wimplicit-fallthrough

### PR DESCRIPTION
**Pull request by qmonnet**
> **Warning** to backporters:
>
> Only the first commit should be backported.

We can label intentional fall-throughs and tell clang to report unannotated ones, to catch bugs that would be caused by involuntarily falling through between switch labels.

The result is not super aesthetic in builtins.h or jhash.h, but at the same time it's just one keyword so not too much boilerplate either. And it catches bugs. The one fixed in commit 9a641116021a (`"bpf: test: fix pktgen for IPv6 NEXTHDR_DEST option"`), or the first commit in this PR, were found this way.

I wasn't able to pin down the exact version that got support for the relevant attribute, but clang 10 has it for sure and it seems to be much older than that, so we don't have to worry and test for its availability.

Inspired by checkpatch complaining about the `/* fall through */` comments.

### CODE REVIEW

File: bpf/lib/egress_policies.h [hunk 1]
```diff
@@ -371,6 +371,7 @@ srv6_create_state_entry(struct __ctx_buff *ctx)
 
 		if (map_update_elem(&SRV6_STATE_MAP6, inner_ips, outer_ips, 0) < 0)
 			return DROP_INVALID;
+		break;
```
Comment (lmb) [inline bpf/lib/egress_policies.h:375]: These seem like genuine mistakes, maybe split them into a separate commit with a Fixes tag?

**Review by lmb — APPROVED** [review]
Nice!

File: bpf/lib/egress_policies.h (hunk 1 above)
Comment (qmonnet) [reply bpf/lib/egress_policies.h:375]: Absolutely, I meant to do that in the first place :facepalm:. Thanks! Will update.

**Review by julianwiedmann — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

**Comment by qmonnet** [comment]
Repeating here the warning for backporters that **only the first commit should be backported to 1.13**.

**Comment by nbusseneau** [comment]
Thanks for the warning @qmonnet!
